staging/src/k8s.io/mount-utils/mount.go

```diff
@@ -147,6 +147,23 @@ func NewMountError(mountErrorValue MountErrorType, format string, args ...interf
 type SafeFormatAndMount struct {
 	Interface
 	Exec utilexec.Interface
+
+	formatSem     chan any
+	formatTimeout time.Duration
+}
+
+// SetMaxConcurrentFormat sets the maximum number of concurrent format
+// operations executed by the mounter. The timeout controls the maximum
+// duration of a format operation before it is treated as complete. A max
+// concurrent value of less than 1 sets concurrency to unlimited.
+// This function assumes that no format operations are currently in flight.
+func (mounter *SafeFormatAndMount) SetMaxConcurrentFormat(n int, timeout time.Duration) {
+	if n > 0 {
+		mounter.formatSem = make(chan any, n)
+		mounter.formatTimeout = timeout
+	} else {
+		mounter.formatSem = nil
```

Comment:
this ignores the timeout parameter and leaves any previously set timeout value in place... is there a reason for that?

Reply:
The concurrency limit is conditional on `formatSem` being set, so the `formatTimeout` value felt inconsequential. No harm in zeroing it out. Done.